src/langcheck/augment/en/_gender/_gender.py

```diff
@@ -0,0 +1,122 @@
+
+from langcheck.augment.en._gender._gender_pronouns import (_PRONOUNS_DICT,
+                                                           _BaseGenderPronouns)
+
+try:
+    nltk.data.find('averaged_perceptron_tagger')
+except LookupError:
+    nltk.download('averaged_perceptron_tagger')
+
+# This dictionary is used to determine the form of the pronoun.
+# Note that his and hers are not included in this dictionary because they can be
+# either of two different forms depending on the context.
+_PRONOUNS_FORM_DICT = MappingProxyType({
+    "she": "subject",
+    "he": "subject",
+    "they": "subject",
+    "their": "dependent_possessive",
+    "him": "object",
+    "them": "object",
+    "hers": "independent_possessive",
+    "theirs": "independent_possessive",
+    "himself": "reflexive",
+    "herself": "reflexive",
+    "themselves": "reflexive"
+})
+
+
+def _get_pronoun_form(word: str, tag: str) -> str | None:
+    """Get pronoun form of the word."""
+    # Handle degenerated cases.
+    if tag == "PRP$":
+        # PRP$ tag denotes a possessive pronoun.
+        return "dependent_possessive"
+    if word.lower() == "her":
+        return "object"
+    if word.lower() == "his":
+        return "independent_possessive"
+    return _PRONOUNS_FORM_DICT.get(word.lower())
+
+
+def _replace_pronoun(word: str, tag: str, target_pronouns: _BaseGenderPronouns):
+    # When the word is not a pronoun, return the word itself.
+    if (pronoun_form := _get_pronoun_form(word, tag)) is None:
+        return word
+    # Replace the pronoun with the target pronoun with the same form.
+    replaced_pronoun = getattr(target_pronouns, pronoun_form)
+    if word.isupper():
+        return replaced_pronoun.upper()
+    elif word.istitle():
+        return replaced_pronoun.title()
+    # When the word is not first letter capitalized or uppercase only, return
+    # the word in lowercase, regardless of how irregularly the word is
+    # capitalized.
+    return replaced_pronoun
+
+
+def _replace_gender_pronouns(
+    text: str,
+    target_pronouns: _BaseGenderPronouns,
+) -> str:
+    """Replace target pronouns in text with new pronouns.
+
+    Args:
+        target_pronouns (_BaseGenderPronouns): Pronouns to replace with.
+        text (str): Text to be augmented.
+
+    Returns:
+        str: Augmented text.
+    """
+    tagged_words = pos_tag(word_tokenize(text))
+    augmented_words = [
+        _replace_pronoun(word, tag, target_pronouns)
+        for (word, tag) in tagged_words
+    ]
+    return TreebankWordDetokenizer().detokenize(augmented_words)
+
+
+def gender(
+    texts: Iterable[str] | str,
+    *,
+    to_gender: str | None = None,
+) -> list[str]:
+    """Replace pronouns with that of specified gender.
+
+        Args:
+            texts: Iterable of texts to be augmented.
+            to_gender: Replacing pronoun type string ('male', 'female' or
+                'neutral'). Default to None and use third-person plural when not
```

Comment:
One drive-by comment: Instead of `None` representing third-person plural, I think we should just set `to_gender = 'plural'` by default. That makes it a bit more explicit. 

(Also wow, I've never heard of these neutral neopronouns before.)

Reply:
Thanks for the comment. I updated the default argument.

Reply:
Thanks!